### ISSUE TITLE
Can add multiple (duplicate or conflicting) conditions

### ISSUE THREAD

**Issue by sssbox** [adamchainz/gargoyle#207]
https://github.com/adamchainz/gargoyle/blob/9d699ae94deccedd35b90b0ce93f48fcb503f9dc/gargoyle/models.py#L141

It seems like this line of code is supposed to prevent adding conditions that already exist, however because `self.value[namespace][field_name]` is a list of tuples with the second item as the condition, the `condition not in` will always be True.

Seems like there are two solutions, either just prevent duplicates:
```python
condition_tuple = (exclude and EXCLUDE or INCLUDE, condition)
if condition_tuple not in self.value[namespace][field_name]:
    self.value[namespace][field_name].append(condition_tuple)
```
or prevent adding conflicting or duplicate conditions (seems like this would probably make more sense, however probably shouldn't just silently fail, should probably let the user know to remove the other condition first):
```python
if condition not in (c for _, c in self.value[namespace][field_name]):
```

I might be able to find some time to work on a patch if nobody else wants to quickly throw something together, but would like some feedback first.

**Comment by adamchainz** [comment]
Thanks for your in-depth analysis. It looks right to me. It also seems you are half-way to a patch so it would be easiest for you to do it, if you can find the time.

**Comment by adamchainz** [comment]
Conflicting would be more useful, but perhaps preventing duplicates would be a good first step.

**Comment by adamchainz** [comment]
Fixed in #212
